packages/@aws-cdk/aws-servicecatalog/lib/portfolio.ts

```diff
@@ -0,0 +1,351 @@
+
+  /**
+   * Add a Launch Notification Constraint.
+   */
+  addEventNotifications(props: EventNotificationsProps): void;
+
+  /**
+   * Add a Launch Role Constraint.
+   */
+  addLaunchRole(props: LaunchRoleProps): void;
+
+  /**
+   * Add a Launch Template Constraint.
+   */
+  addProvisioningRules(props: ProvisioningRulesProps): void;
+
+  /**
+   * Add a Resource Update Constraint.
+   */
+  allowTagUpdates(props: TagUpdatesProps): void;
+
+  /**
+   * Add a Stack Set Constraint.
+   */
+  addStackSetConstraint(props: StackSetConstraintProps): void;
+
+  /**
+   * Associate Tag Options
+   */
+  addTagOptions(tagOptions: TagOption): void;
+}
+
+/**
+ * A reference to a Service Catalog portfolio.
+ */
+export interface PortfolioAttributes {
+
+  /**
+   * The ARN of the portfolo.
+   */
+  readonly portfolioArn: string;
+
+  /**
+   * The name of the portfolo.
+   */
+  readonly portfolioName: string;
+}
+
+/**
+ * Represents a Service Catalog portfolio.
+ */
+abstract class PortfolioBase extends Resource implements IPortfolio {
+
+  /**
+   * The ARN of the portfolio.
+   */
+  public abstract readonly portfolioArn: string;
+
+  /**
+   * The Id of the portfolio.
+   */
+  public abstract readonly portfolioId: string;
+
+  /**
+   * The name of the portfolio.
+   */
+  public abstract readonly portfolioName: string;
+
+  /**
+   * Enable the Portfolio for a specific Role
+   */
+  public giveAccessToRole(role: iam.IRole) {
+    this.associatePrincipal(role.roleArn, role.node.addr);
+  }
+
+  /**
+   * Enable the Portfolio for a specific User
+   */
+  public giveAccessToUser(user: iam.IUser) {
+    this.associatePrincipal(user.userArn, user.node.addr);
+  }
+
+  /**
+   * Enable the Portfolio for a specific Group
+   */
+  public giveAccessToGroup(group: iam.IGroup) {
+    this.associatePrincipal(group.groupArn, group.node.addr);
+  }
+
+  /**
+   * Associate a Service Catalog product to the portfolio.
+   */
+  public addProduct(product: IProduct) {
+    AssociationManager.associateProductWithPortfolio(this, this, product);
+  }
+
+  /**
+   * Share the portfolio with a designated account.
+   */
+  public share(accountId: string, shareTagOptions?: boolean, acceptLanguage?: AcceptLanguage) {
+    const hashId = this.getKeyForPortfolio(accountId);
+    new CfnPortfolioShare(this, `PortfolioShare${hashId}`, {
+      portfolioId: this.portfolioId,
+      accountId: accountId,
+      shareTagOptions: shareTagOptions,
+      acceptLanguage: acceptLanguage,
+    });
+  }
+
+  /**
+   * Associate a principal with the portfolio.
+   */
+  private associatePrincipal(principalArn: string, identifier: string) {
+    const hashId = this.getKeyForPortfolio(identifier);
+    new CfnPortfolioPrincipalAssociation(this, `PortolioPrincipalAssociation${hashId}`, {
+      portfolioId: this.portfolioId,
+      principalArn: principalArn,
+      principalType: PrincipalType.IAM,
+    });
+  }
```

Comment:
`private` methods should be below all public ones (I'm surprised our linter allowed this).